pika/heartbeat.py

```diff
@@ -9,37 +9,32 @@
 
 class HeartbeatChecker(object):
     """Checks to make sure that our heartbeat is received at the expected
-    intervals.
+    timeouts.
 
     """
-    DEFAULT_INTERVAL = 60
-    MAX_IDLE_COUNT = 2
-
+    # Note: even though we're sending heartbeats in half the specified
+    # timeout value, the broker will be sending them to us at the specified
+    # value. This means we'll be checking for an idle connection via send_and_check
+    # twice as many times as the broker will send heartbeats to us,
+    # so we need to set max idle count to 4 here
+    _MAX_IDLE_COUNT = 4
+    _CONNECTION_FORCED = 320
     _STALE_CONNECTION = "Too Many Missed Heartbeats, No reply in %i seconds"
 
-    def __init__(self, connection, interval=DEFAULT_INTERVAL, idle_count=MAX_IDLE_COUNT):
+    def __init__(self, connection, timeout):
         """Create a heartbeat on connection sending a heartbeat frame every
-        interval seconds.
+        timeout seconds.
 
         :param pika.connection.Connection: Connection object
-        :param int interval: Heartbeat check interval. Note: heartbeats will
-                             be sent at interval / 2 frequency.
-        :param int idle_count: The number of heartbeat intervals without data
-                               received that will close the current connection.
+        :param int timeout: Heartbeat check timeout. Note: heartbeats will
+                            actually be sent at timeout / 2 frequency.
 
         """
         self._connection = connection
 
         # Note: see the following document:
         # https://www.rabbitmq.com/heartbeats.html#heartbeats-timeout
-        self._interval = float(interval / 2)
-
-        # Note: even though we're sending heartbeats in half the specified
-        # interval, the broker will be sending them to us at the specified
-        # interval. This means we'll be checking for an idle connection
-        # twice as many times as the broker will send heartbeats to us,
-        # so we need to double the max idle count here
-        self._max_idle_count = idle_count * 2
+        self._timeout = float(timeout / 2)
```

Comment:
The math counts don't work out with this PR: division by 2 versus _MAX_IDLE_COUNT. `self._timeout` is just timeout without any division. Cast it to a float, though, to make division correct on python 2.x later when calculating the delay value to pass to _adapter_add_timeout(). In other words: 
```
self._timeout = float(timeout)
```

Or, if you want to store pre-computed "check interval delay", then 
```
self._check_interval = float(timeout) / HeartbeatChecker._MAX_IDLE_COUNT
```
Note that having float wrapping `timeout` turns the division into floating point op on python 2.x too.

In subsequent comments, I will assume you are using `self._check_interval`

Reply:
Great, `_check_interval` is a much better name.